tests/circuit_graph/test_circuit_graph_hash.py

```diff
@@ -0,0 +1,752 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit and integration tests for creating the :mod:`pennylane` :attr:`QNode.circuit.circuit_hash` attribute.
+"""
+import pytest
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Tensor
+from pennylane.circuit_graph import CircuitGraph
+from pennylane.qnodes import BaseQNode
+from pennylane.variable import Variable
+
+class TestCircuitGraphHash:
+    """Test the creation of a hash to the CircuitGraph"""
+
+    numeric_queues = [
+                        ([
+                            qml.RX(0.3, wires=[0])
+                        ],
+                         [],
+                        'RX!0.3![0]|||'
+                        ),
+                        ([
+                            qml.RX(0.3, wires=[0]),
+                            qml.RX(0.4, wires=[1]),
+                            qml.RX(0.5, wires=[2]),
+                        ],
+                         [],
+                        'RX!0.3![0]RX!0.4![1]RX!0.5![2]|||'
+                        )
+                     ]
+
+    @pytest.mark.parametrize("queue, observable_queue, expected_string", numeric_queues)
+    def test_create_hash_argument_numeric_arguments(self, queue, observable_queue, expected_string):
+        """Tests that the same hash is created for two circuitgraphs that have numeric arguments."""
+        circuit_graph_1 = CircuitGraph(queue + observable_queue, {})
+        circuit_graph_2 = CircuitGraph(queue + observable_queue, {})
+
+        assert circuit_graph_1.create_hash_argument() == circuit_graph_2.create_hash_argument()
+        assert expected_string == circuit_graph_1.create_hash_argument()
+
+
+    variable = Variable(1)
+
+    symbolic_queue = [
+                        ([qml.RX(variable, wires=[0])],
+                         [],
+                        'RX!' + str(variable.idx) + '![0]|||'
```

Comment:
In this case, `variable.idx` is just `1` right?

Reply:
True! Would it be better to read as the exact string in there?

Reply:
Just double checking! It should be fine. Another approach would be something like
```python
    symbolic_queue = [
                        ([qml.RX(Variable(1), wires=[0])],
                         [],
                        'RX!1![0]|||'
```
so that you're comparing a completely symbolic circuit queue against a fully serialized result

Reply:
True, thanks! The reason, why I had ``Variable(1)`` separated is because there are certain tests that make use of having the same ``Variable`` object used and so wanted to pass the same object. I'll modify the expected serialized string though!